ansible/roles/common-cartridge/tasks/main.yml

```
@@ -8,14 +8,38 @@
   tags:
     - configmap
 
+- name: Remove External Secret
+  kubernetes.core.k8s:
+    kubeconfig: ~/.kube/config
+    namespace: "{{ NAMESPACE }}"
+    template: external-secret.yml.j2
+    state: absent
+  when:
+    - EXTERNAL_SECRETS_OPERATOR is defined and EXTERNAL_SECRETS_OPERATOR|bool
+    - WITH_BRANCH_MONGO_DB_MANAGEMENT is undefined or not WITH_BRANCH_MONGO_DB_MANAGEMENT|bool
+  tags:
+    - 1password
+
 - name: 1Password
   kubernetes.core.k8s:
     kubeconfig: ~/.kube/config
     namespace: "{{ NAMESPACE }}"
     template: onepassword.yml.j2
   when: 
-      - ONEPASSWORD_OPERATOR is defined and ONEPASSWORD_OPERATOR|bool
-      - WITH_COMMON_CARTRIDGE is defined and WITH_COMMON_CARTRIDGE|bool
+    - ONEPASSWORD_OPERATOR is defined and ONEPASSWORD_OPERATOR|bool
+    - WITH_COMMON_CARTRIDGE is defined and WITH_COMMON_CARTRIDGE|bool
+  tags:
+    - 1password
+
+- name: External Secret (namespace specific)
+  kubernetes.core.k8s:
+    kubeconfig: ~/.kube/config
+    namespace: "{{ NAMESPACE }}"
+    template: external-secret.yml.j2
+  when:
+    - EXTERNAL_SECRETS_OPERATOR is defined and EXTERNAL_SECRETS_OPERATOR|bool
+    - WITH_BRANCH_MONGO_DB_MANAGEMENT is defined and WITH_BRANCH_MONGO_DB_MANAGEMENT|bool
+    - WITH_COMMON_CARTRIDGE is defined and WITH_COMMON_CARTRIDGE|bool

```

I think if we use the `WITH_BRANCH_MONGO_DB_MANAGEMENT` and the thing with `state` we cann have only one yob for the External Secrets
```suggestion
- name: External Secret (namespace specific)
  kubernetes.core.k8s:
    kubeconfig: ~/.kube/config
    namespace: "{{ NAMESPACE }}"
    template: external-secret.yml.j2
    state: "{{ 'present' if WITH_BRANCH_MONGO_DB_MANAGEMENT else 'absent'}}
  when:
    - EXTERNAL_SECRETS_OPERATOR is defined and EXTERNAL_SECRETS_OPERATOR|bool
    - WITH_COMMON_CARTRIDGE is defined and WITH_COMMON_CARTRIDGE|bool

```

---

Yes, that works and is cleaner. I changed it in both PRs.